### PR TITLE
GPII-3096: Update code coverage to use nyc instead of istanbul.

### PR DESCRIPTION
**Pull request by the-t-in-rtf**
See [GPII-3096](https://issues.gpii.net/browse/GPII-3096) for details.

### CODE REVIEW

**Comment by amb26** [comment]
I tried this out and it worked fine, but the coverage statistics reported on the console were for a wider set of files than those in the HTML coverage report at gpii-express/reports/lcov-report/ . This isn't particularly a problem, but we should probably make these consistent. The former also showed that there is a lot of code in gpii-express/tests/js/lib/diagramAllRoutes.js which seems to be never executed.
We may as well update to grunt 1.0.3 while we are at it as the only outdated dependency.

File: package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
     "main": "index.js",
     "scripts": {
         "pretest": "node node_modules/rimraf/rimraf coverage/* reports/*",
```
Comment (amb26) [inline package.json]: It turns out that this rimraf line silently does nothing. Execute rimraf/bin.js instead

Comment (the-t-in-rtf) [reply package.json]: Good catch.  Fixed.

**Comment by the-t-in-rtf** [comment]
@amb26, I updated the NYC config, you can see an updated report here:

http://the-t-in-rtf.github.io/coverage-reports/gpii-express/20180607/

The test content was not meant to be instrumented or reported on.  I can see making an exception for some of the bits that tend to be used in other work, but in that case we should move those parts to src/test and add docs.  I eventually plan to purge the sequence mangling bits in this package in favour of proper sequence steps, so I'd rather not advertise those at all.  If you think it's worth it, I can move the diagramming bit into src/test and add separate tests.

**Comment by amb26** [comment]
That's fine, I have just the same coverage holes in Kettle's test utilities for the same reasons

**Pull request merged**